cmd/swarmctl/cluster/update.go

```diff
@@ -114,6 +119,11 @@ var (
 				if err != nil {
 					return err
 				}
+
+				if hbPeriod < time.Duration(0) {
```

Comment:
This is a bug and needs to be fixed. https://github.com/docker/swarmkit/issues/1018 has been filed.